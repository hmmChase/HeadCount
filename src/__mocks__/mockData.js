export const mockCard1 = {
  location: 'COLORADO',
  average: 0.53,
  stats: {
    '2004': 0.24,
    '2005': 0.278,
    '2006': 0.337,
    '2007': 0.395,
    '2008': 0.536,
    '2009': 0.598,
    '2010': 0.64,
    '2011': 0.672,
    '2012': 0.695,
    '2013': 0.703,
    '2014': 0.741
  }
};

export const mockCard2 = {
  location: 'ACADEMY 20',
  average: 0.407,
  stats: {
    '2004': 0.302,
    '2005': 0.267,
    '2006': 0.354,
    '2007': 0.392,
    '2008': 0.385,
    '2009': 0.39,
    '2010': 0.436,
    '2011': 0.489,
    '2012': 0.479,
    '2013': 0.488,
    '2014': 0.49
  }
};

export const mockCard3 = {
  location: 'ADAMS COUNTY 14',
  average: 0.709,
  stats: {
    2004: 0.228,
    2005: 0.3,
    2006: 0.293,
    2007: 0.306,
    2008: 0.673,
    2009: 1,
    2010: 1,
    2011: 1,
    2012: 1,
    2013: 0.998,
    2014: 1
  }
};

export const comparisonObject = {
  'ACADEMY 20': 0.407,
  'ADAMS COUNTY 14': 0.709,
  compared: 0.574
};
