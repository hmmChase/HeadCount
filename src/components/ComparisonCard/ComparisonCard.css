.comparisonCard {
  border-radius: 10px;
  background-color: #406287;
  border: 1px solid #e7ecefad;
  cursor: pointer;
  width: 40vh;
  margin: 2%;
  max-height: 50vh;
  -webkit-box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  box-shadow: 0 4px 8px 0 rgb(162, 168, 171);
  -webkit-transition: 0.3s;
  -o-transition: 0.3s;
  transition: 0.3s;
  -webkit-animation-duration: 2s;
  animation-duration: 3s;
  -webkit-animation-name: card-opacity;
  animation-name: card-opacity;
  color: white;
  padding-top: 10vh;
}
