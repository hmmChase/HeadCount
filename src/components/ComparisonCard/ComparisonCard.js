import React from 'react';
import PropTypes from 'prop-types'
import './ComparisonCard.css'

const ComparisonCard = ({comparisonObject}) => {
  const keys = Object.keys(comparisonObject)
  return (
    <article className='comparisonCard'>
      <h4>{keys[0]} : {comparisonObject[keys[0]]} </h4> 
      <h4>{keys[1]} : {comparisonObject[keys[1]]} </h4>

      <h3>Comparison: {comparisonObject[keys[2]]}</h3>
    </article>
  )
}

ComparisonCard.propTypes = {
  comparisonObject: PropTypes.object.isRequired
}

export default ComparisonCard;