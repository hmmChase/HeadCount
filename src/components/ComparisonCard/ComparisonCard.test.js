import React from 'react';
import { shallow } from 'enzyme';
import ComparisonCard from './ComparisonCard';
import { comparisonObject } from '../../__mocks__/mockData';

const mockProps = {
  comparisonObject: comparisonObject
};

describe('ComparisonContainer', () => {
  let container;
  beforeEach(() => {
    container = shallow(<ComparisonCard {...mockProps} />);
  });

  it('matches snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
