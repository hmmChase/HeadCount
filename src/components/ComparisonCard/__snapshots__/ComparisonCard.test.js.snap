// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonContainer matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonCard
    comparisonObject={
        Object {
            "ACADEMY 20": 0.407,
            "ADAMS COUNTY 14": 0.709,
            "compared": 0.574,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          ACADEMY 20
           : 
          0.407
           
</h4>,
        <h4>
          ADAMS COUNTY 14
           : 
          0.709
           
</h4>,
        <h3>
          Comparison: 
          0.574
</h3>,
      ],
      "className": "comparisonCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "ACADEMY 20",
            " : ",
            0.407,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "ACADEMY 20",
          " : ",
          0.407,
          " ",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "ADAMS COUNTY 14",
            " : ",
            0.709,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "ADAMS COUNTY 14",
          " : ",
          0.709,
          " ",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Comparison: ",
            0.574,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Comparison: ",
          0.574,
        ],
        "type": "h3",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            ACADEMY 20
             : 
            0.407
             
</h4>,
          <h4>
            ADAMS COUNTY 14
             : 
            0.709
             
</h4>,
          <h3>
            Comparison: 
            0.574
</h3>,
        ],
        "className": "comparisonCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "ACADEMY 20",
              " : ",
              0.407,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "ACADEMY 20",
            " : ",
            0.407,
            " ",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "ADAMS COUNTY 14",
              " : ",
              0.709,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "ADAMS COUNTY 14",
            " : ",
            0.709,
            " ",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Comparison: ",
              0.574,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Comparison: ",
            0.574,
          ],
          "type": "h3",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
